### PR TITLE
Add `format` subcommand to xtask

### LINKED ISSUE #1518: Add rustfmt xtask subcommand
The way we use cargo workspaces means that rustfmt doesn't format all of our projects. We should extend the xtask to handle this.

### PR DESCRIPTION
**Pull request by playfulFence**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
#### Description
Closes #1518 

#### Testing
Added some unformatted code and executed `cargo xtask fmt-packages`

### CODE REVIEW

File: esp-hal/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `time::current_time` API (#1503)
 - ESP32-S3: Add LCD_CAM Camera driver (#1483)
 - `embassy-usb` support (#1517)
+- Added `fmt-packages` xtask subcommand (#1551)
```
Comment (SergioGasquez) [inline esp-hal/CHANGELOG.md:20]: Does this deserve a esp-hal changelog entry?? I think I would just `skip-changelog`

Comment (jessebraham) [reply esp-hal/CHANGELOG.md:20]: It does not, I am not sure why I need to keep repeating this but:

__Changes to the `esp-hal/CHANGELOG.md` file are only required if there are user-facing changes to the `esp-hal` package__.

**Review by jessebraham — APPROVED** [review]
Not really sure why this is needed but looks fine to me I guess.

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

**Comment by SergioGasquez** [comment]
HIL CI is failing due to https://github.com/knurling-rs/defmt/issues/834

**Pull request merged**